### ISSUE TITLE
ArgumentOutOfRangeException (Parameter name: start) when loading image

### ISSUE THREAD

**Issue by mabead** [SixLabors/ImageSharp#694]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [x] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
Loading image throws exception.

### Steps to Reproduce
With 1.0.0-beta005, calling Image.Load on the attached file throws the following exception:

```
System.ArgumentOutOfRangeException
  HResult=0x80131502
  Message=Specified argument was out of the range of valid values.
  Source=System.Memory
  StackTrace:
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument)
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifReader.TryReadSpan(Int32 length, ReadOnlySpan`1& span)
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifReader.ReadUInt16()
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifReader.AddValues(List`1 values, Int32 index)
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifReader.GetThumbnail(Int32 offset)
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifReader.ReadValues()
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifProfile.InitializeValues()
   at SixLabors.ImageSharp.MetaData.Profiles.Exif.ExifProfile.TryGetValue(ExifTag tag, ExifValue& value)
   at SixLabors.ImageSharp.Formats.Jpeg.JpegDecoderCore.InitDerivedMetaDataProperties()
   at SixLabors.ImageSharp.Formats.Jpeg.JpegDecoderCore.Decode[TPixel](Stream stream)
   at SixLabors.ImageSharp.Formats.Jpeg.JpegDecoder.Decode[TPixel](Configuration configuration, Stream stream)
   at SixLabors.ImageSharp.Image.Decode[TPixel](Stream stream, Configuration config)
   at SixLabors.ImageSharp.Image.WithSeekableStream[T](Configuration config, Stream stream, Func`2 action)
   at SixLabors.ImageSharp.Image.Load[TPixel](Configuration config, Stream stream, IImageFormat& format)
   at SixLabors.ImageSharp.Image.Load(Stream stream)
```

This can be reproduced with these images.

[OutOfRange.zip](https://github.com/SixLabors/ImageSharp/files/2357298/OutOfRange.zip)

### System Configuration

- ImageSharp version: 1.0.0-beta005
- Other ImageSharp packages and versions: 
- Environment (Operating system, version and so on): windows
- .NET Framework version: .NET Core 2.0
- Additional information:

**Comment by JimBobSquarePants** [comment]
>With 1.0.0-beta0003, calling Image.Load on the attached file throws the following exception:

1.0.0-beta0003?

**Comment by mabead** [comment]
Sorry, typo. It was beta005. I updated the description.

**Comment by JimBobSquarePants** [comment]
Fixed with #699
